src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -1103,17 +1103,27 @@ class AbstractOption(models.Model):
     for a particular item.  Instead, option need to be specified by a customer
     when they add the item to their basket.
     """
-    name = models.CharField(_("Name"), max_length=128)
-    code = AutoSlugField(_("Code"), max_length=128, unique=True,
-                         populate_from='name')
 
-    REQUIRED, OPTIONAL = ('Required', 'Optional')
+    # Option types
+    TEXT = "text"
+    INTEGER = "integer"
+    FLOAT = "float"
+    BOOLEAN = "boolean"
+    DATE = "date"
+
     TYPE_CHOICES = (
-        (REQUIRED, _("Required - a value for this option must be specified")),
-        (OPTIONAL, _("Optional - a value for this option can be omitted")),
+        (TEXT, _("Text")),
+        (INTEGER, _("Integer")),
+        (BOOLEAN, _("True / False")),
+        (FLOAT, _("Float")),
+        (DATE, _("Date")),
     )
-    type = models.CharField(_("Status"), max_length=128, default=REQUIRED,
-                            choices=TYPE_CHOICES)
+
+    name = models.CharField(_("Name"), max_length=128)
+    code = AutoSlugField(_("Code"), max_length=128, unique=True,
+                         populate_from='name')
+    type = models.CharField(_("Type"), max_length=255, default=TEXT, choices=TYPE_CHOICES)
```

Comment:
Since we are changing this field anyway, let's avoid using reserved words. Prefer `option_type` instead of `type`.

Reply:
Tried it and a bit too much work at this point.